### ISSUE TITLE
static/icon.png hash is not invalidated with `target: static`

### ISSUE THREAD

**Issue by Josh979** [nuxt-community/pwa-module#353]
The app icon images that are generated seem to use the default/initial icon.png image for icon generation when using the default icon.png filename. I tried different configurations to get it to refresh, and I ended up having to build once using an alternate filename, and then switch the filename back to the original, then build again, in order to allow the new icon.png file to be used.

Nuxt version: 2.14.5 and 2.14.6
This occurred in both @nuxt/pwa v3.0.0-beta20 and v3.0.2

Initial run. No options added to config. I updated icon.png file in static dir folder. Result: The new icon.png image was not used to generate icon files, instead the old icon.png file was used. 

```  
pwa:{  },

```
Next attempt: Setting the values in the options also did not use new icon.png image, still the old icon.png image was used.
```  
  pwa:{
    icon:{
      source: '/static/icon.png',
      fileName: 'icon.png'
    },
  },

```

Finally got a new image to be used for generation by first changing the filename+source to a different file,
```  
  pwa:{
    icon:{
      source: '/static/icon-alt.png',
      fileName: 'icon-alt.png'
    },
  },

```

I then changed the icon config back to the default filename, added a new icon.png file to the static directory, ran build, and the new icon.png file was used correctly.
```  
  pwa:{
    icon:{
      source: '/static/icon.png',
      fileName: 'icon.png'
    },
  },

```

**Comment by leopoldkristjansson** [comment]
I was struggling with this too. I found out that leaving the `/` before `static/...` worked for me.

```
  pwa: {
    icon: {
      source: 'static/icon.png',
      filename: 'icon.png'
    },
  },
```

Now the icon _builds_ again.

**Comment by Josh979** [comment]
@leopoldkristjansson Thanks for the tip, I'll have to try that out in the meantime.

**Comment by ponnex** [comment]
> I was struggling with this too. I found out that leaving the `/` before `static/...` worked for me.
> 
> ```
>   pwa: {
>     icon: {
>       source: 'static/icon.png',
>       filename: 'icon.png'
>     },
>   },
> ```
> 
> Now the icon _builds_ again.

This works for me thanks!

**Comment by crytos** [comment]
Hello, guys is there a way of the icon.source being an external image URL, please. I am working on a dynamic subdomain kind of app and need to generate different icons basing on the logo of the current subdomain app

`pwa: {
    icon: {
      source: 'https://res.cloudinary.com/viera/image/upload/v1599731538/uploads/it6atyi5fbnquuxiyoqk.png',
      filename: 'And what do i put here please'
    },
  },
 `

Thank you, please.

**Comment by ponnex** [comment]
> Hello, guys is there a way of the icon.source being an external image URL, please. I am working on a dynamic subdomain kind of app and need to generate different icons basing on the logo of the current subdomain app
> 
> `pwa: { icon: { source: 'https://res.cloudinary.com/viera/image/upload/v1599731538/uploads/it6atyi5fbnquuxiyoqk.png', filename: 'And what do i put here please' }, }, `
> 
> Thank you, please.

I think it isn't possible with this module, as it generates icons on manifest.json. The closes I could find is this [article](https://medium.com/@rajathans/dynamic-pwa-manifest-using-javascript-and-rest-api-a23d132de7d6)

**Comment by crytos** [comment]
Hey, @ponnex Let me check out that link, please. Thank you

**Comment by studnitz** [comment]
The icon is cached in `rootDir/node_modules/.cache/pwa/icons`. Clearing that directory gets the new icon, still a bug though iMo.

**Comment by crytos** [comment]
> The icon is cached in `rootDir/node_modules/.cache/pwa/icon`. Clearing that directory gets the new icon, still a bug though iMo.

Yeah, that's right. Actually, I just added an icons array in the manifest option abject. and disabled the icons module.

**Comment by crytos** [comment]
Hey @ponnex  Later I figured that my problem is bigger. I am trying to build an app like "StudioYou" - the one you referred in the article above, It is called myduuka.com, It's a platform for creating online shops, whereby each user can have an independent online shop on their subdomain. 

So I have been trying to make a request to the server according to the current host(subdomain) so that I can get the data for that particular subdomain. But since to my knowledge, I can't access the window.location.host (subdomain) from the nuxt.config, I am stuck there  currently

**Comment by pi0** [comment]
Hi @Josh979. I'm trying to investigate this issue but cache key is generated on file contents ([src](https://github.com/nuxt-community/pwa-module/blob/b0f226aadeee50b47790ac064852b444e33d9b95/lib/icon/module.js#L129)). Are you using `target: static` or `nuxt generate`?

**Comment by pi0** [comment]
@crytos You can ask your generic questions in [nuxtjs](https://github.com/nuxt/nuxt.js/issues) or discord :)

**Comment by Josh979** [comment]
> Hi @Josh979. I'm trying to investigate this issue but hash for cache is generated based on file contents ([src](https://github.com/nuxt-community/pwa-module/blob/b0f226aadeee50b47790ac064852b444e33d9b95/lib/icon/module.js#L129)). Are you using `target: static` or `nuxt generate`? 

@pi0 Yes, I use both.

**Comment by pi0** [comment]
Ahh i see. This is a bug with nuxt [smart rebuild](https://nuxtjs.org/blog/nuxt-static-improvements/) as we don't hash [static dir](https://github.com/nuxt/nuxt.js/blob/dev/packages/cli/src/utils/generate.js#L36) and icon module will be skipped. You can try `nuxt generate --force-build` but will look for a better way :)

**Comment by pi0** [comment]
Hi. Sorry for inconveniences. This issue should be resolved by [v3.2.0](https://github.com/nuxt-community/pwa-module/releases/tag/v3.2.0).

**Comment by larsdouweschuitema** [comment]
> I was struggling with this too. I found out that leaving the `/` before `static/...` worked for me.
> 
> ```
>   pwa: {
>     icon: {
>       source: 'static/icon.png',
>       filename: 'icon.png'
>     },
>   },
> ```
> 
> Now the icon _builds_ again.

This worked for me, thank you. 💯
